tinygrad/runtime/ops_cuda.py

```diff
@@ -46,8 +46,10 @@ def __call__(self, global_size, local_size, *args, wait=False):
 
 class CUDACodegen(CStyleCodegen):
   lang = CStyleLanguage(
-    kernel_prefix = "__global__", smem_prefix = "__shared__ ", barrier = "__syncthreads();", float4 = "make_float4",
+    kernel_prefix = "using int64 = int64_t; using uchar = unsigned char;\n__global__", smem_prefix = "__shared__ ",
+    barrier = "__syncthreads();", float4 = "make_float4",
     half_prekernel = "#include <cuda_fp16.h>",
+    float_conversion = {x.name: "__float2uint_rn" if dtypes.is_unsigned(x) else "__float2int_rn" for x in [dtypes.int8, dtypes.int32, dtypes.int64, dtypes.uint8]},
```

Comment:
can dtypes.is_int() be used here ?

Reply:
Came up `{x.name: "__float2uint_rn" if dtypes.is_unsigned(x) else "__float2int_rn" for x in map(lambda x: x.default, fields(dtypes)) if dtypes.is_int(x)}`, but requires another line for import

I say we leave it, as similar direct reference is also done in `llvmir.py`. If need arises, maybe an `all_dtypes` function will be added in the future.